### PR TITLE
add unit test for cli

### PR DESCRIPTION
**Pull request by chloeyin**
Signed-off-by: zaihaoyin <zaihaoyin@microsoft.com>

This PR adds unit test for some cli commands, only to make sure the flags are parsed correctly.
This is the last PR of the whole series(https://github.com/notaryproject/notation/pull/255).

### CODE REVIEW

**Comment by JeyJeyGao** [comment]
LGTM

File: cmd/notation/cache_test.go [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+	cmd := cachePruneCommand(nil)
+	if err := cmd.ParseFlags([]string{}); err != nil {
+		t.Fatalf("Parse Flag failed: %v", err)
+	}
+	if err := cmd.Args(cmd, cmd.Flags().Args()); err == nil {
+		t.Fatal("Parse Args expected error, but ok")
+	}
+}
+
+func TestCacheRemoveCommand(t *testing.T) {
```
Comment (binbin-li) [inline cmd/notation/cache_test.go:76]: Could you specify the testing case like line 107? Same to other tests.

File: cmd/notation/cache_test.go [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+	}
+	if err := cmd.Args(cmd, cmd.Flags().Args()); err != nil {
+		t.Fatalf("Parse Args failed: %v", err)
+	}
+	if !reflect.DeepEqual(*expected, *opts) {
+		t.Fatalf("Expect cache remove opts: %v, got: %v", expected, opts)
+	}
+}
+
+func TestCacheRemoveCommand_MissinArgs(t *testing.T) {
```
Comment (binbin-li) [inline cmd/notation/cache_test.go:107]: typo: Missing

**Comment by binbin-li** [comment]
Left some tiny comments. Just wanna raise some questions out of scope.
1. Looks like in this PR we focus on the main use cases, successful case and a common failed case which should cover most of cases. But users can fail a command in a lot ways (a combination of flags or args). It's kind of time-wasting to test all of them, just wanna confirm we would do the same to the new PRs. @shizhMSFT 
2. This PR tests the conversion from command input to the internal `Opts` object. For the business logics after it, is there a good way to test it? Maybe just run `cmd.ParseFlags()` like this PR does? Or move the logics to other places for easier testing?

**Comment by binbin-li** [comment]
lgtm overall

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Comment by shizhMSFT** [comment]
> Left some tiny comments. Just wanna raise some questions out of scope.
> 
> 1. Looks like in this PR we focus on the main use cases, successful case and a common failed case which should cover most of cases. But users can fail a command in a lot ways (a combination of flags or args). It's kind of time-wasting to test all of them, just wanna confirm we would do the same to the new PRs. @shizhMSFT
> 2. This PR tests the conversion from command input to the internal `Opts` object. For the business logics after it, is there a good way to test it? Maybe just run `cmd.ParseFlags()` like this PR does? Or move the logics to other places for easier testing?

We are still figuring out a better way to test CLI.

**Pull request merged**